src/main/groovy/com/google/protobuf/gradle/GenerateProtoTask.groovy

```diff
@@ -343,11 +344,12 @@ public abstract class GenerateProtoTask extends DefaultTask {
     state = State.FINALIZED
   }
 
-  @Internal("Tracked as an input via getDescriptorSetOptionsForCaching()")
+  @OutputFile
+  @PathSensitive(PathSensitivity.RELATIVE)
+  @Optional
```

Comment:
Why not change `DescriptorSetOptions.path`'s annotations instead? That is still marked as an `@Input`.

Reply:
Thanks @lptr. Fixed